independent-projects/arc/runtime/src/main/java/io/quarkus/arc/InterceptionProxy.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.arc;
+
+/**
+ * A factory for proxies that perform {@link jakarta.interceptor.AroundInvoke AroundInvoke}
+ * interception before forwarding the method call to the target instance ({@code delegate}).
+ * <p>
+ * This construct only supports {@code @AroundInvoke} interceptors declared on
+ * interceptor classes; other kinds of interception, as well as {@code @AroundInvoke}
+ * interceptors declared on the target class and its superclasses, are not supported.
+ * <p>
+ * The container provides a built-in bean with scope {@link jakarta.enterprise.context.Dependent Dependent}
+ * and qualifier {@link jakarta.enterprise.inject.Default Default} that can be injected.
```

Comment:
```suggestion
 * and qualifier {@link jakarta.enterprise.inject.Default Default} that can be injected into method parameters of a producer method.
```

And we should probably also mention that synthetic beans can obtain a factory using the `SyntheticCreationalContext#getInterceptionProxy()` method.

Reply:
Done.